### ISSUE TITLE
Error adding ssl wsdl

### ISSUE THREAD

**Issue by nanndoj** [nordic-institute/X-Road#292]
I am getting the following error trying to add the wsdl `https://sca.indea.mt.gov.br/SIA/webservices/SCAServices.jws?wsdl`.

<img width="542" alt="Screenshot 2019-12-12 at 14 16 03" src="https://user-images.githubusercontent.com/8084261/70715205-13e63380-1cea-11ea-80b0-576ff542ca2f.png">

It works fine with the qa version: `http://jao.cepromat.mt.gov.br:8180/SIA/webservices/SCAServices.jws?wsdl`

I believe the difference is in the header:

**Not working (Note the https in the definition)**
```
<wsdl:definitions targetNamespace="https://sca.indea.mt.gov.br/SIA/webservices/SCAServices.jws" xmlns:apachesoap="https://xml.apache.org/xml-soap" xmlns:impl="https://sca.indea.mt.gov.br/SIA/webservices/SCAServices.jws" xmlns:intf="https://sca.indea.mt.gov.br/SIA/webservices/SCAServices.jws" xmlns:soapenc="https://schemas.xmlsoap.org/soap/encoding/" xmlns:wsdl="https://schemas.xmlsoap.org/wsdl/" xmlns:wsdlsoap="https://schemas.xmlsoap.org/wsdl/soap/" xmlns:xsd="https://www.w3.org/2001/XMLSchema">
```

**Working (QA)**
```
<wsdl:definitions xmlns:apachesoap="http://xml.apache.org/xml-soap" xmlns:impl="http://jao.cepromat.mt.gov.br:8180/SIA/webservices/SCAServices.jws" xmlns:intf="http://jao.cepromat.mt.gov.br:8180/SIA/webservices/SCAServices.jws" xmlns:soapenc="http://schemas.xmlsoap.org/soap/encoding/" xmlns:wsdl="http://schemas.xmlsoap.org/wsdl/" xmlns:wsdlsoap="http://schemas.xmlsoap.org/wsdl/soap/" xmlns:xsd="http://www.w3.org/2001/XMLSchema" targetNamespace="http://jao.cepromat.mt.gov.br:8180/SIA/webservices/SCAServices.jws">
```

Is that how it is supposed to be? Any workaround?

**Comment by petkivim** [comment]
Hi @nanndoj 

In xml namespaces, the URI (in this case an URL, http://schemas.xmlsoap.org/wsdl/) is the unique name for the namespace. The namespace URI does not need to be an URL, and it does not need to point to anything, although that is a convention. When used to refer to a schema, it must be the same as defined in the schema's targetNameSpace attibute, or the definition won't match the elements defined in the schema. In the WSDL schema:

```
<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema"
           xmlns:wsdl="http://schemas.xmlsoap.org/wsdl/"
           targetNamespace="http://schemas.xmlsoap.org/wsdl/"
           elementFormDefault="qualified" >
```

The declaration `xmlns:wsdl="https://schemas.xmlsoap.org/wsdl/"` defines that in this xml document, the prefix `wsdl` means `https://schemas.xmlsoap.org/wsdl/`, which is not the target namespace of the elements defined in the WSDL schema. For example, the element `<wsdl:definitions>` is expanded to a fully-qualified name `{https://schemas.xmlsoap.org/wsdl/}definitions)` which does not match the expected `{http://schemas.xmlsoap.org/wsdl/}definitions` causing validation error.

In other words, the problem can be fixed by updating the declaration (`xmlns:wsdl="https://schemas.xmlsoap.org/wsdl/"`) in your production WSDL from `https` to `http`.

Best regards,
Petteri

**Comment by petkivim** [comment]
In addition, the WSDL description is not aligned with the X-Road Message Protocol for SOAP. The WSDL must include the X-Road specific SOAP headers and other elements. More detailed information can be foud here:

https://github.com/nordic-institute/X-Road/blob/develop/doc/Protocols/pr-mess_x-road_message_protocol.md#3-describing-services

Best regards,
Petteri

**Comment by nanndoj** [comment]
Thanks @petkivim for the detailed explanation. I'll close this issue now
